pkg/kubelet/pod_manager.go

```diff
@@ -110,7 +126,37 @@ func (self *basicPodManager) UpdatePods(u PodUpdate, podSyncTypes map[types.UID]
 	}
 }
 
-func updatePods(changed []api.Pod, current []api.Pod) []api.Pod {
+// Need to acquire the lock before calling this function.
+func (self *basicPodManager) SetPods(newPods []api.Pod) {
+	pods := []api.Pod{}
```

Comment:
Less allocations which is in general faster, not a huge concern for speed. My main concern is that we take the address of an element in an expanding slice. When the slice gets re-allocated and moved I *think* Go still keeps the old one around since we have a pointer going into it, but now you have at least two copies of parts of the same data:

http://play.golang.org/p/GOTmm6KmZ5